src/main/java/org/apache/sysds/runtime/instructions/cp/UnaryScalarCPInstruction.java

```diff
@@ -26,10 +26,10 @@
 import org.apache.sysds.runtime.matrix.operators.Operator;
 import org.apache.sysds.runtime.matrix.operators.UnaryOperator;
 
-public class UnaryScalarCPInstruction extends UnaryMatrixCPInstruction {
+public class UnaryScalarCPInstruction extends UnaryCPInstruction {
```

Comment:
This is a problem due to inheritance. We replace `UnaryMatrixCPInstruction`s, but don't replace `UnaryScalarCPInstruction`s, with FED instructions. I think this inheritance made no sense and probably was a bug.
If it *wasn't* we can also overload the `tryReplaceWithFederated()` here.